yaksh/views.py

```diff
@@ -3237,3 +3237,51 @@ def download_course_progress(request, course_id):
     for student in stud_details:
         writer.writerow(student)
     return response
+
+
+@login_required
+@email_verified
+def download_course_yaml(request, course_id):
+    user = request.user
+    course = Course.objects.get(id=course_id)
+    if not is_moderator(user):
+        raise Http404('You are not allowed to download this course !')
+    if not course.is_creator(user):
+        raise Http404('This course does not belong to you')
+
+    zip_file_path = course.create_yaml()
+
+    response = HttpResponse(content_type='application/zip')
+    response['Content-Disposition'] = dedent(
+        '''attachment; filename={0}_yaml.zip'''.format(course.name)
```

Comment:
filename is the course name
course name may contain spaces, better to replace them by underscores

Reply:
OK, Done using slugify because even though it uses '-' instead of '_' it does a better job of handling special characters and multiple consecutive spaces